bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/AirMediaSink.java

```diff
@@ -111,46 +113,69 @@ public String getId() {
             throws UnsupportedAudioFormatException, UnsupportedAudioStreamException {
-        if (thingHandler.getThing().getStatus() == ThingStatus.ONLINE) {
+        if (thingHandler.getThing().getStatus() != ThingStatus.ONLINE) {
+            tryClose(audioStream);
+            return;
+        }
+
+        if (audioStream == null) {
+            stopMedia();
+            return;
+        }
+
+        String url;
+        if (audioStream instanceof URLAudioStream urlAudioStream) {
+            // it is an external URL, we can access it directly
+            url = urlAudioStream.getURL();
+            tryClose(audioStream);
+        } else {
+            // we serve it on our own HTTP server
+            logger.debug("audioStream {} {}", audioStream.getClass().getSimpleName(), audioStream.getFormat());
+            StreamServed streamServed;
             try {
-                MediaReceiverManager manager = thingHandler.getManager(MediaReceiverManager.class);
-                if (audioStream == null) {
-                    manager.sendToReceiver(playerName, password, Action.STOP, MediaType.VIDEO);
-                    return;
-                }
-
-                if (audioStream instanceof URLAudioStream urlAudioStream) {
-                    // it is an external URL, we can access it directly
-                    logger.debug("AirPlay audio sink: process url {}", urlAudioStream.getURL());
-                    playMedia(manager, urlAudioStream.getURL());
-                    return;
-                }
-                // we serve it on our own HTTP server
-                StreamServed streamServed;
-                try {
-                    streamServed = audioHTTPServer.serve(audioStream, 5, true);
-                } catch (IOException e) {
-                    try {
-                        audioStream.close();
-                    } catch (IOException ex) {
-                        logger.debug("Exception while closing audioStream");
-                    }
-                    throw new UnsupportedAudioStreamException(
-                            "AirPlay device was not able to handle the audio stream (cache on disk failed).",
-                            audioStream.getClass(), e);
-                }
-                streamServed.playEnd().thenRun(() -> this.playbackFinished(audioStream));
-                logger.debug("AirPlay audio sink: process url {}", callbackUrl + streamServed.url());
-                playMedia(manager, callbackUrl + streamServed.url());
-            } catch (FreeboxException e) {
-                logger.warn("Audio stream playback failed: {}", e.getMessage());
+                streamServed = audioHTTPServer.serve(audioStream, 5, true);
+            } catch (IOException e) {
+                tryClose(audioStream);
+                throw new UnsupportedAudioStreamException(
+                        "AirPlay device was not able to handle the audio stream (cache on disk failed).",
```

Comment:
AirPlay intended? Just checking.

Reply:
Yes, it concerns Freebox devices (Freebox players) having AirPlay support.